lib/ui/src/modules/ui/configs/handle_routing.js

```diff
@@ -92,8 +92,22 @@ export default function({ clientStore }, actions) {
   // handle initial URL
   handleInitialUrl(actions, window.location);
 
+  const data = clientStore.getAll();
+  let prevKind = data.selectedKind;
+  let prevStory = data.selectedStory;
+
   // subscribe to clientStore and change the URL
-  clientStore.subscribe(() => changeUrl(clientStore));
+  clientStore.subscribe(() => {
+    const { selectedKind, selectedStory } = clientStore.getAll();
+    // use pushState only when a new story is selected
+    const usePush =
+      prevKind != null &&
```

Comment:
Is there something to push the state for the first story storybook loads when there isn't a previous kind? Or is that not really necessary?

Reply:
Imagine that we opened storybook by a link from other website. If we'll create additional history entries on initialization stage, we won't be able to return to referring website by just hitting "back" button.

That's why we need to replace not push in this case

Reply:
gotcha, makes sense